### DISCUSSION TITLE
Generate model interfaces and keep in source control for sharing (without client)?

### DISCUSSION THREAD

**Discussion by basememara** [prisma/prisma#20677] [Q&A] [answered]
### Question

I'd like to share my model interfaces with other apps and packages, but obviously not share the Prisma Client or provide access to my database connection. My idea is to have Prisma generate interfaces for the models that would be stored in source control for sharing. That way, callers can send API request in a safe and strongly-typed way.

Is there any way to achieve this or will this require a deep change in the Prisma source code?

**Comment by nurul3101** [comment]
Hello @basememara 👋 

You can create a private npm package which exports all Prisma generated types, which you can then use in your other apps and packages. I have seen numerous users try out this approach to share the types.

Here's an example that could be helpful:
https://github.com/igolka97/shared-prisma-client-example

Would that work for you?

**Comment by basememara** [comment]
That would be perfect! I couldn't find this in the docs or an example project, anything that comes to mind? So far, I'm thinking to have a post script in my build process but I don't know what to do from there.

**Reply by nurul3101** [reply]
Does this example help?
https://github.com/igolka97/shared-prisma-client-example

**Reply by basememara** [reply]
Unfortunately I do not want to share the client, but just the types. I'm currently sharing the Prisma package in my monorepo, but now I want to move my UI components in a separate 3rd party package but do want to give them access to my database or the whole client, just the model types.

Ideally, the UI components would not have a dependency or even know about Prisma. They would just know about the generated model types in a separate package that has nothing else in it.

My thought is maybe I can have a post build script that would copy just the types from the Prisma directory to this blank, dedicated model types package, but the Prisma types seem pretty melted into the stack.

Any way to achieve just grabbing the model types only and copy them somewhere else alone instead of the whole Prisma infrastructure and client along with it?

**Comment by basememara** [comment]
It looks like you answered a StackOverflow that suggests to use the `index.d.ts` file, but it looks like you still need Prisma as a dependency but at least do not need the client generated nor the database connection (?).

Leaving it here as a reference: https://stackoverflow.com/questions/73852913/share-prisma-types-and-enums-across-multiple-projects

**Reply by basememara** [reply, marked as answer]
This looks like it worked! In my post migration script, I'm copying the definition file to my app to store in the repo. No prisma dependency needed in my app:

```
// monorepo package.json > scripts
"db:migrate": "yarn workspace @acme/database prisma migrate dev",
"postdb:migrate": "cp ./node_modules/.prisma/client/index.d.ts ./apps/my-app/types/Prisma.d.ts"
```
